src/io/hdf/hdf_path.cpp

```diff
@@ -70,7 +80,18 @@ hdf_path& hdf_path::replace_subgroup(std::string_view p)
   return append(p);
 }
 
-bool hdf_path::has_root_directory() const { return (!path_.empty() && path_[0] == '/'); }
```

Comment:
The old code seems solid to me. Based on C/C++ rule, if the first condition failed, the second should not run.

Reply:
I'm also struggling to understand why short-circuit evaluation isn't working here.
https://en.wikipedia.org/wiki/Short-circuit_evaluation